lib/dialects/mysql/query-generator.js

```diff
@@ -265,6 +266,16 @@ var QueryGenerator = {
 
     if (attribute.references) {
       attribute = Utils.formatReferences(attribute);
+
+      if (options && options.context === 'addColumn' && options.foreignKey) {
+        var attrName = options.foreignKey;
+
+        template += Utils._.template(', ADD CONSTRAINT <%= fkName %> FOREIGN KEY (<%= attrName %>)')({
+          fkName: this.quoteIdentifier(attrName + '_foreign_idx'),
```

Comment:
Can you point to where you found that naming? It looks like create tables leaves out a name (so it uses the database generated one).

If there's no existing naming please consider making `[table]_[field]_fkey`.